spark3-runtime/LICENSE

```diff
@@ -579,3 +579,19 @@ License text:
 | NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 | SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+--------------------------------------------------------------------------------
+
+This binary artifact contains code from Project Nessie.
+
+Copyright: 2020 Dremio Corporation.
+Home page: https://projectnessie.org/
+License: http://www.apache.org/licenses/LICENSE-2.0
+
+--------------------------------------------------------------------------------
+
+This binary artifact contains Eclipse Jersey bindings for jackson.
```

Comment:
EPL 2.0 is listed under [Category B, "weak" copyleft licenses](https://www.apache.org/legal/resolved.html). The relevant sections from that page are:

> Software under the following licenses may be included in binary form within an Apache product if the inclusion is appropriately labeled (see above)

> By including only the object/binary form, there is less exposed surface area of the third-party work from which a work might be derived; this addresses the second guiding principle of this policy.

And here's the part about labelling:

> APPROPRIATELY LABELLED CONDITION
> In all Category B cases our users should not be surprised at their inclusion in our products. By attaching an appropriate and prominent label to the distribution users are less likely to be unaware of restrictions significantly different from those of the Apache License. An appropriate and prominent label is a label the user will read while learning about the distribution - for example in a README, and it should identify the third-party product, its licensing, and a url to the its homepage. Please also ensure to comply with any attribution/notice requirements in the specific license in question.

I think that means we will need a "prominent label" that a user will read while learning about the distribution. I would add one as a new section to the project's README, as well as on the releases page. We should also link to it from anywhere we recommend downloading a runtime Jar on the site. We should also add any notes needed to comply with [EPL's requirements](https://www.eclipse.org/legal/epl-2.0/#requirements).

Is it possible to remove this dependency? That might be easier.

Reply:
Somehow I apparently didn't post and lost the comments that I thought I had added to this ticket. @rymurr we should look at how other tools handle this. Jersey is used by most Apache projects and most of it is already packaged by tools like Spark, Hive, etc. The only thing we're actually doing here is adding the extension for jackson encoding/decoding for Jersey. 

@rdblue we're also exploring avoiding jersey entirely as Hive 2.x has a nasty old version of it that is causing us headaches.

Reply:
FWIW Spark distributes [LICENSE-binary](https://github.com/apache/spark/blob/master/LICENSE-binary) and [NOTICE-binary](https://github.com/apache/spark/blob/master/NOTICE-binary) with a footnote for any non-apache licenses.

Hopefully I can remove jersey entirely and skip EPL completely